tests/testthat/test-expect-self-test.R

```diff
@@ -1,45 +1,78 @@
-test_that("fail always fails", {
-  expect_failure(fail())
-  expect_failure(fail("abc"), "abc")
-})
+test_that("expect_failure() requires 1 failure and zero successes", {
+  expect_success(expect_failure(fail()))
 
-test_that("succeed always succeeds", {
-  expect_success(succeed())
+  expect_failure(expect_failure({}))
+  expect_failure(expect_failure(succeed()))
+  expect_failure(expect_failure({
```

Comment:
a bit torn but worth considering adding the parallel test with two `fail()`s, even though the second can't be reached?

Reply:
Agreed, I think that's a good place to reinforce that we should only ever see one fail